src/main/java/org/commcare/core/interfaces/VirtualDataInstanceCache.java

```diff
@@ -0,0 +1,20 @@
+package org.commcare.core.interfaces;
+
+import org.javarosa.core.model.instance.VirtualDataInstance;
+
+import java.sql.SQLException;
+import java.util.UUID;
+
+import javax.annotation.Nullable;
+
+/**
+ * Read and write operations for entity selections made on a mult-select Entity Screen
+ */
+public interface VirtualDataInstanceCache {
+
+    UUID write(VirtualDataInstance dataInstance);
```

Comment:
Is there a specific reason why we anchor on UUID objects here in particular?

For better or worse, we are fairly consistent in the other parts of the app in simply keying on strings that happen to be uuid's, and I'm worried about mixing the datatypes resulting in a sprawling set of implicit casts.

Reply:
This has been carried forward from the actual [postgres model](https://github.com/dimagi/formplayer/blob/multiSelectList/src/main/java/org/commcare/formplayer/db/migration/V23__init_virtual_data_instance.java#L13) itself. Are you suggesting that we do a type conversion before writing to the Postgres or change the datatype in db itself ?

Reply:
My inclination would be to change datatype itself and just key with a unique indexed string to stay consistent with other GUID indexes in the system. 

It's less efficient, for sure, than a proper DB UUID but I think the odds of us running into an issue with having a mix of string represented UUID's and typed UUID's are pretty high. It might be an interesting experiment to try to use strongly typed UUID's here to see if we can start shifting, but it would be a bad outcome if we only saw the edges late enough that we ended up casting UUID's <-> Strings

Reply:
https://github.com/dimagi/commcare-core/pull/1109/commits/e888ecf7e6afe5285d7be8d84d725c691c47462f and https://github.com/dimagi/formplayer/pull/1139/commits/1f3c5b392be59fa69aca53391dfdb56e73a95b44